lib/src/tasks/test/cli.dart

```diff
@@ -164,7 +175,38 @@ class TestCli extends TaskCli {
     // Build the list of tests to run.
     if (individualTestsSpecified) {
       // Individual tests explicitly passed in should override the test suites.
-      tests.addAll(parsedArgs.rest);
+      if (dartMajorVersion == 2 && parsedArgs[_hackFastBuilds]) {
+        reporter.warning(
+            'You\'re using `${_hackFastBuilds}`. This will re-write the generated test runners in your repo.\n'
+                'The test task will attempt to restore your generated runners after completion, but you may '
+                'have to re-run `pub run dart_dev gen-test-runner` if your runners have changed.');
+        final mapConfigToTestFiles = <TestRunnerConfig,
+            List<String> /* tests to include in runner */>{};
+        // Construct mapping from config to tests which should be ran in that config
+        for (final _config in config.genTestRunner.configs) {
+          for (final testFilePath in parsedArgs.rest) {
```

Comment:
Would there be some issue if someone also passed the `--release` flag instead of or at the end of end of a list of files?

Reply:
`.rest` includes anything which wasn't specified as a top level option/flag, so we should be good here.